src/auction.js

```diff
@@ -661,14 +662,13 @@ export function getStandardBidderSettings(mediaType, bidderCode) {
         }
     };
   }
+
   const TARGETING_KEYS = CONSTANTS.TARGETING_KEYS;
+  const standardSettings = Object.assign({}, bidderSettings.settingsFor(null));
 
-  let bidderSettings = $$PREBID_GLOBAL$$.bidderSettings;
-  if (!bidderSettings[CONSTANTS.JSON_MAPPING.BD_SETTING_STANDARD]) {
-    bidderSettings[CONSTANTS.JSON_MAPPING.BD_SETTING_STANDARD] = {};
-  }
-  if (!bidderSettings[CONSTANTS.JSON_MAPPING.BD_SETTING_STANDARD][CONSTANTS.JSON_MAPPING.ADSERVER_TARGETING]) {
-    bidderSettings[CONSTANTS.JSON_MAPPING.BD_SETTING_STANDARD][CONSTANTS.JSON_MAPPING.ADSERVER_TARGETING] = [
+  if (!standardSettings[CONSTANTS.JSON_MAPPING.ADSERVER_TARGETING]) {
+    standardSettings[CONSTANTS.JSON_MAPPING.ADSERVER_TARGETING] = [
+      // NOTE: some callers (probably tests) are modifying these - they are not safe to cache
```

Comment:
I tried generating these once as the original version did, but someone is messing with the return value. I was not able to find who that is or why that was not a problem before my change.

Reply:
This was something I recently fixed.

https://github.com/prebid/Prebid.js/pull/7607

The default way this function is called is through https://github.com/prebid/Prebid.js/blob/4c424a06883a421bb25dd96d1cf8fbda90741e9c/src/auction.js#L674

This always passes undefined. And for these KVP sets it needs access to the bidReq.mediaType.

So when KVP's are set, it actually always runs the OR `getPriceGranularity(bid.mediaType, bidReq);`

Which will in turn select the right granularity.

There are only two other places which use this function, and they directly pass in the granularity, which is why we had to support the `granularity || granularity`

https://github.com/prebid/Prebid.js/blob/master/modules/adpod.js#L126

https://github.com/prebid/Prebid.js/blob/master/modules/konduitWrapper.js#L104

So this is actually fine with the current way this is setup, but yours is probably smarter.

Also the entire Targeting logic is crazy and needs to be rewritten! haha.

Reply:
I don't think it's fine in general: the `|| getPriceGranularity(bid.mediaType, bidReq)` is only evaluated on the first call because `granularity` is in the closure scope. See the test case I added in `auctionmanager_spec`, that fails without this change.

Reply:
I get it. Thanks!